strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/io/RepositoryFileSystemProvider.java

```diff
@@ -371,17 +447,49 @@ public void checkAccess(Path path,
     public <A extends BasicFileAttributes> A readAttributes(Path path,
                                                             Class<A> type,
                                                             LinkOption... options)
-            throws IOException
+        throws IOException
     {
-        return storageFileSystemProvider.readAttributes(getTargetPath(path), type, options);
+        A targetAttributes = storageFileSystemProvider.readAttributes(getTargetPath(path), type, options);
+        if (!(path instanceof RepositoryPath))
+        {
+            return targetAttributes;
+        }
+        RepositoryPath repositoryPath = (RepositoryPath) path;
+        RepositoryPath repositoryRelativePath = repositoryPath.getRepositoryRelative();
+        
+        RepositoryFileAttributes repositoryFileAttributes = new RepositoryFileAttributes(targetAttributes,
+                getRepositoryFileAttributes(repositoryRelativePath));
+        return (A) repositoryFileAttributes;
     }
 
+    protected abstract Map<String,Object> getRepositoryFileAttributes(RepositoryPath repositoryRelativePath);
+
+
+    public boolean isChecksum(RepositoryPath path)
+    {
+        for (String e : path.getFileSystem().getDigestAlgorithmSet())
+        {
+            if (path.endsWith("." + e.replaceAll("-", "").toLowerCase()))
```

Comment:
it creates Checksum file extension name form Digest algorithm name: `SHA-1`->`sha1`